Code/src/main/java/nl/utwente/group10/ui/components/Connection.java

```diff
@@ -175,18 +182,22 @@ public final void changed(ObservableValue<? extends Number> observable,
 
 	public final void disconnect(ConnectionAnchor anchor) {
 		if (startAnchor.isPresent() && startAnchor.get().equals(anchor)) {
-			startAnchor.get().disconnect(this);
+			startAnchor.get().disconnectFrom(this);
 			startAnchor = Optional.empty();
 		}
 		if (endAnchor.isPresent() && endAnchor.get().equals(anchor)) {
-			endAnchor.get().disconnect(this);
+			endAnchor.get().disconnectFrom(this);
 			endAnchor = Optional.empty();
 		}
 	}
+	
+	public final void disconnect(Optional<? extends ConnectionAnchor> anchor){
+		disconnect(anchor.orElse(null));
+	}
 
 	public final void disconnect() {
-		disconnect(startAnchor.orElse(null));
-		disconnect(endAnchor.orElse(null));
+		disconnect(startAnchor);
+		disconnect(endAnchor);
```

Comment:
Van wat ik uit de docs begrijp is de Optional.equals() een normale equals methode, en is het pas gelijk als de andere ook een Optional is.